core/amber/src/main/scala/edu/uci/ics/texera/web/resource/auth/UserResource.scala

```diff
@@ -27,21 +52,38 @@ object UserResource {
 
   private def setUserSession(session: HttpSession, user: User): Unit = {
     session.setAttribute(SESSION_USER, user)
+  }
 
+  private def setGoogleUserSession(session: HttpSession, Googleuser: GoogleUser): Unit = {
+    session.setAttribute(SESSION_GOOGLE_USER, Googleuser)
   }
+
 }
 
 @Path("/users/")
 @Consumes(Array(MediaType.APPLICATION_JSON))
 @Produces(Array(MediaType.APPLICATION_JSON))
 class UserResource {
 
+  val config: Config = WebUtils.config
+
+  private val CLIENT_ID = config.getString("google.clientId")
+  private val CLIENT_SECRET = config.getString("google.clientSecret")
+
   final private val userDao = new UserDao(SqlServer.createDSLContext.configuration)
+  final private val googleUserDao = new GoogleUserDao(SqlServer.createDSLContext.configuration)
+
+  private val TRANSPORT = new NetHttpTransport
+  private val JSON_FACTORY = new JacksonFactory
 
   @GET
   @Path("/auth/status")
-  def authStatus(@Session session: HttpSession): Option[User] = {
-    getUser(session)
+  def authStatus(@Session session: HttpSession) = {
```

Comment:
now as we are adding `google-user` into this, it is not clear what Type is being returned in this function. Is it possible add type annotation here? or it will just result into different JSON objects (serialized user/google-user)?

Reply:
It will just result into different JSON objects.